src/main/java/com/google/step/snippet/external/StackOverflowClient.java

```diff
@@ -69,18 +63,17 @@ public Card search(String url) {
       return null;
     }
     String title = getTitle(questionId);
-    String answerBody = getAnswerBody(answerId);
-    if (title == null || answerBody == null) {
+    String description = getAnswerBody(answerId);
+    if (title == null || description == null) {
       return null;
     }
-    // No description or code is allowed for StackOverflow.
-    String description = getDescription(answerBody);
-    String code = getCode(answerBody);
+    // Code is intentionally set to null for better display.
+    String code = null;
     return new Card(title, code, url, description);
   }
 
   /* Get the question id of passed in URL. */
-  private String getQuestionId(String url) {
+  String getQuestionId(String url) {
     URI uri;
     try {
       uri = new URI(url);
```

Comment:
new URI can also throw a NullPointerException when url is null, although i guess URL should never be null if its gotten this far.

Reply:
url will not be null. if it's null, the search servlet will skip searching method for this client.